File: tests/fixtures/csharp/simple.cs
/**
 * FIXTURE: Simple C# class
 * TESTS: Basic method signature extraction
 */

using System;
using System.Collections.Generic;

namespace MyApp
{
    public class UserService
    {
        private readonly ILogger _logger;

        public UserService(ILogger logger)
        {
            _logger = logger;
        }

        public async Task<User> GetUser(int id)
        {
            var user = await _repository.FindById(id);
            if (user == null)
                throw new NotFoundException($"User {id} not found");
            return user;
        }

        public void DeleteUser(int id)
        {
            _repository.Delete(id);
            _logger.Info($"Deleted user {id}");
        }
    }
}

File: tests/fixtures/ruby/simple.rb
# FIXTURE: Simple Ruby class
# TESTS: Basic method signature extraction

require 'json'
require 'net/http'

class UserService
  attr_reader :logger

  def initialize(logger)
    @logger = logger
  end

  def find_user(id)
    user = User.find(id)
    raise NotFoundError, "User #{id} not found" unless user
    user
  end

  def delete_user(id)
    User.destroy(id)
    logger.info("Deleted user #{id}")
  end

  private

  def validate_id(id)
    raise ArgumentError, "Invalid ID" unless id.positive?
  end
end
